server/controllers/auth.js

```diff
@@ -177,13 +179,17 @@ function loadSessionInformation(user) {
       [session.user] = rows;
 
       // next make sure this user has permissions
+      // we use now roles for assigning permissions to users
       sql = `
-        SELECT IF(permission.user_id = ?, 1, 0) authorized, unit.path
-        FROM unit LEFT JOIN permission
-          ON unit.id = permission.unit_id;
+        SELECT IF(user_role.user_id = ?, 1, 0) authorized, unit.path
+        FROM unit 
+        LEFT JOIN role_unit ON unit.id = role_unit.unit_id
+        LEFT JOIN role ON role.uuid = role_unit.role_uuid
```

Comment:
Do you need the `LEFT JOIN role`?   It looks like this will work equally well:
```sql
SELECT IF(user_role.user_id = ?, 1, 0) authorized, unit.path
FROM unit
LEFT JOIN role_unit ON unit.id = role_unit.unit_id
LEFT JOIN user_role ON user_role.role_uuid =role_unit.role_uuid
WHERE user_role.user_id = ?;
```

It would save MySQL a bit of work if it does!